chainer/functions/connection/depthwise_convolution_2d.py

```diff
@@ -1,124 +1,4 @@
-            w_type.ndim == 4,
-            x_type.shape[1] == w_type.shape[1],
-        )
-
-        if type_check.eval(n_in) == 3:
-            b_type = in_types[2]
-            type_check.expect(
-                b_type.dtype == x_type.dtype,
-                b_type.ndim == 1,
-                b_type.shape[0] == w_type.shape[0] * w_type.shape[1],
-            )
-
-    def forward(self, inputs):
-        x, W = inputs[:2]
-        b = inputs[2] if len(inputs) == 3 else None
-        kh, kw = W.shape[2:]
-
-        xp = cuda.get_array_module(*x)
-        if xp is numpy:
-            self.col = conv.im2col_cpu(
-                x, kh, kw, self.sy, self.sx, self.ph, self.pw)
-        else:
-            self.col = conv.im2col_gpu(
-                x, kh, kw, self.sy, self.sx, self.ph, self.pw)
-
-        B, C, KY, KX, IY, IX = self.col.shape
-        D = W.shape[0]  # (D, C, KY, KX)
-        c_ = self.col.transpose(1, 0, 4, 5, 2, 3) \
-            .reshape((C, B * IY * IX, KY * KX))
-        w_ = W.transpose(1, 2, 3, 0).reshape((C, KY * KX, D))
-
-        # (C, B*IY*IX, KY*KX), (C, KY*KX, D)-> (C, B*IY*IX, D)
-        y = _matmul(c_, w_, xp).astype(x.dtype, copy=False)
-
-        # (C, B*IY*IX, D) -> (B, C*D, IY, IX)
-        y = y.reshape((C, B, IY * IX, D)).transpose(1, 0, 3, 2) \
-            .reshape((B, C * D, IY, IX))
-
-        if b is not None:
-            y += b[None, :, None, None]
-        return y,
-
-    def backward(self, inputs, grad_outputs):
-        x, W = inputs[:2]
-        b = inputs[2] if len(inputs) == 3 else None
-        gy = grad_outputs[0]
-        h, w = x.shape[2:]
-
-        xp = cuda.get_array_module(*x)
-
-        B, C, KY, KX, IY, IX = self.col.shape
-        D = W.shape[0]
-
-        # (B, C*D, IY, IX) -> (C, D, B*IY*IX, D)
-        gy_ = gy.reshape((B, C, D, IY * IX)).transpose(1, 2, 0, 3) \
-            .reshape((C, D, B * IY * IX))
-        c_ = self.col.transpose(1, 0, 4, 5, 2, 3) \
-            .reshape((C, B * IY * IX, KY * KX))
-        # (C, D, B*IY*IX), (C, B*IY*IX, KY*KX) -> (C, D, KY*KX)
-        gW_ = _matmul(gy_, c_, xp)
-        gW = gW_.reshape((C, D, KY, KX)).transpose(1, 0, 2, 3)
-        gW = gW.astype(W.dtype, copy=False)
-
-        w_ = W.transpose(1, 2, 3, 0).reshape((C, KY * KX, D))
-        # (C, KY*KX, D), (C, D, B*IY*IX) -> (C, KY*KX, B*IY*IX)
-        gcol = _matmul(w_, gy_, xp).reshape((C, KY, KX, B, IY, IX))
-        gcol = gcol.astype(x.dtype, copy=False)
-        gcol = xp.rollaxis(gcol, 3)
-
-        if xp is numpy:
-            gx = conv.col2im_cpu(gcol, self.sy, self.sx,
-                                 self.ph, self.pw, h, w)
-        else:
-            gx = conv.col2im_gpu(gcol, self.sy, self.sx,
-                                 self.ph, self.pw, h, w)
-
-        if b is None:
-            return gx, gW
-        else:
-            gy = xp.rollaxis(gy, 1, 4)
-            gb = gy.sum(axis=(0, 1, 2))
-            return gx, gW, gb
+from chainer import functions
```

Comment:
Fix circular import (python<3.5) by importing the module `chainer.functions.connection.convolution_2d`.

Reply:
fixed by only importing `chainer`